src/utils.c

```diff
@@ -201,9 +201,10 @@ time_value_to_internal(Datum time_val, Oid type)
 
 		return DatumGetInt64(res);
 	}
-
-	elog(ERROR, "unkown time type oid '%d'", type);
-	return -1;
+	/* time column must be binary compatible with bigint if it is not one of the above types
+	 * this check takes place in dimension.c when creating a table
+	 */
+	return DatumGetInt64(time_val);
 }
```

Comment:
This function is used in other places as well to convert a datatype given in SQL query to internal type (such as `drop_chunks`).
It is important to detect when an incompatible type is passed from SQL so I think it makes more sense to have an error in default case and maybe add some custom logic to test if the type is custom binary.

Reply:
Ah, I was not aware of that. Though I wonder if we shouldn't split the function then - one for external functions for checking and casting and one for casting without checking. Currently to check if it's binary compatible it has to do some lookups in the PG cast table. I am not sure what the overhead would be of always checking this in this function.